numba/cpython/unicode.py

```diff
@@ -842,6 +830,56 @@ def _adjust_indices(length, start, end):
     return start, end
 
 
+@overload_method(types.UnicodeType, 'startswith')
+def unicode_startswith(s, prefix, start=None, end=None):
+    if not (start is None or isinstance(start, (types.Omitted,
+                                                types.Integer,
+                                                types.NoneType))):
+        raise TypingError('When specified, the arg "start" must be an Integer')
+
+    if not (end is None or isinstance(end, (types.Omitted,
+                                            types.Integer,
+                                            types.NoneType))):
+        raise TypingError('When specified, the arg "end" must be an Integer')
+
+    if isinstance(prefix, (types.Tuple, types.UniTuple)):
+        def startswith_tuple_impl(s, prefix, start=None, end=None):
+            for item in prefix:
+                if s.startswith(item, start, end):
+                    return True
+            return False
+
+        return startswith_tuple_impl
+
+    if isinstance(prefix, types.UnicodeCharSeq):
+        def startswith_char_seq_impl(s, prefix, start=None, end=None):
+            return s.startswith(str(prefix), start, end)
+
+        return startswith_char_seq_impl
+
+    if isinstance(prefix, types.UnicodeType):
+        def startswith_unicode_impl(s, prefix, start=None, end=None):
+            length = len(s)
+            sub_length = len(prefix)
+            if start is None:
+                start = 0
+            if end is None:
+                end = length
+
+            start, end = _adjust_indices(length, start, end)
+            if end - start < sub_length:
+                return False
+
+            if sub_length == 0:
+                return True
+
+            s = s[start:end]
+
+            return _cmp_region(s, 0, prefix, 0, sub_length) == 0
```

Comment:
```suggestion
            s_slice = s[start:end]

            return _cmp_region(s_slice, 0, prefix, 0, sub_length) == 0
```
just to make it clear that this is addressing a slice of the original string.

Reply:
OK.